pandaserver/dataservice/event_picker.py

```diff
@@ -204,26 +222,26 @@ def run(self):
             tmpDN = CoreUtils.get_id_from_dn(self.userDN)
             # make dataset container
             tmpRet = self.pd2p.register_dataset_container_with_datasets(
-                self.userDatasetName,
+                self.user_dataset_name,
                 allFiles,
                 locationMap,
                 n_sites=eventPickNumSites,
                 owner=tmpDN,
             )
             if not tmpRet:
-                self.endWithError(f"Failed to make a dataset container {self.userDatasetName}")
+                self.endWithError(f"Failed to make a dataset container {self.user_dataset_name}")
                 return False
             # skip DaTRI
             if skipDaTRI:
```

Comment:
You can remove this line, unindent the if block, and delete https://github.com/PanDAWMS/panda-server/blob/1458c6607bc49cfc28b6a2887c2562f7849a149d/pandaserver/dataservice/event_picker.py#L240-L335

Reply:
Thank you Tadashi, I will also split the big run function into smaller functions